parlai/scripts/self_chat.py

```diff
@@ -86,15 +98,31 @@ def _run_self_chat_episode(opt, world, world_logger):
 
 def self_chat(opt):
     random.seed(opt['seed'])
+    partner = opt['partner_model_file']
+    partner_opt_file = opt['partner_opt_file']
 
     # Create agents
     agent1 = create_agent(opt, requireModelExists=True)
-    agent2 = agent1.clone()
+    if not partner:
+        # Self chat with same model
+        agent2 = agent1.clone()
+    else:
+        # Self chat with different models
+        if partner_opt_file:
+            print(f"WARNING: Loading model with opts from {partner_opt_file}")
+            with open(partner_opt_file) as f:
+                partner_opt = json.load(f)
+            for k, v in partner_opt.items():  # override partner opts
+                opt[k] = v
```

Comment:
ooo actually i just realized we don't want to override this way. instead, you take the loaded override opts from `partner_opt` and instead of `create_agent` let's call 

`create_agent_from_model_file(partner, partner_opt)` <-- this will load the opts from the partner's model file and then allow us to override anything we want to change with `partner_opt`

Reply:
the only argument that we may want to pull from `opt` is 'interactive_mode', so maybe you can carve out a special exception for this

Reply:
(replace code from lines 115-119)

Reply:
re: earlier comment about carving out an exception: `partner_opt['interactive_mode'] = opt['interactive_mode']` (and add a warning that you're doing this)